lib/src/r_tree/non_leaf_node.dart

```diff
@@ -91,13 +91,13 @@ class NonLeafNode<E> extends Node<E> {
     _minimumBoundingRect = null;
   }
 
-  Node<E> _getBestNodeForInsert(RTreeDatum<E> item) {
+  Node<E>? _getBestNodeForInsert(RTreeDatum<E>? item) {
```

Comment:
I think we can rewrite this to return non-null since non-leaf nodes are guaranteed to have at least one item _childNodes (otherwise the calling code would crash, also we convert to a leaf node if child nodes becomes empty):

```dart
Node<E> _getBestNodeForInsert(RTreeDatum<E> item) {
    Node<E> bestNode = _childNodes.first;
    num bestCost = bestNode.expansionCost(item);

    for (var child in _childNodes.skip(1)) {
      final tentativeCost = child.expansionCost(item);
      if (tentativeCost < bestCost) {
        bestCost = tentativeCost;
        bestNode = child;
      }
    }

    return bestNode;
  }
```